src/overlays/actors/ovl_En_Mag/z_en_mag.c

```diff
@@ -27,21 +33,553 @@ const ActorInit En_Mag_InitVars = {
+                                               &D_801333E8);
+
+                        gSaveContext.gameMode = 2;
+                        globalCtx->sceneLoadFlag = 20;
+                        globalCtx->fadeTransition = 2;
+                    }
+
+                    this->copyrightAlphaStep = 15;
+                    this->fadeOutAlphaStep = 25;
+                    this->globalState = MAG_STATE_FADE_OUT;
+                }
+            } else {
+                sDelayTimer--;
+            }
+        }
+    }
+
+    if (this->globalState == MAG_STATE_FADE_IN) {
+        if (this->effectFadeInState == 0) {
+            this->effectAlpha += 6.375f;
+            this->effectPrimLodFrac += 0.8f;
+
+            this->effectPrimColor[0] += 6.375f;
+            this->effectPrimColor[1] += 3.875f;
+            this->effectPrimColor[2] += 2.125f;
+            this->effectEnvColor[0] += 6.375f;
+            this->effectEnvColor[1] += 3.875f;
+
+            this->effectFadeInTimer--;
+
+            if (this->effectFadeInTimer == 0) {
+                this->effectPrimLodFrac = 32.0f;
+                this->effectAlpha = 255.0f;
+
+                this->effectPrimColor[0] = 255.0f;
+                this->effectPrimColor[1] = 255.0f;
+                this->effectPrimColor[2] = 255.0f;
+                this->effectEnvColor[0] = 255.0f;
+                this->effectEnvColor[1] = 255.0f;
+
+                this->effectFadeInTimer = 40;
+                this->effectFadeInState = 1;
+            }
+        } else if (this->effectFadeInState == 1) {
+            this->effectPrimColor[0] += -2.125f;
+            this->effectEnvColor[0] += -1.375f;
+
+            this->effectPrimLodFrac += 2.4f;
+
+            this->effectFadeInTimer--;
+
+            if (this->effectFadeInTimer == 0) {
+                this->effectPrimLodFrac = 128.0f;
+
+                this->effectPrimColor[0] = 170.0f;
+                this->effectEnvColor[0] = 200.0f;
+
+                this->effectFadeInTimer = 32;
+                this->effectFadeInState = 2;
+            }
+        }
+
+        if (this->effectAlpha > 160) {
+            this->mainAlpha += VREG(5);
+            if (this->mainAlpha >= 210.0f) {
+                this->mainAlpha = 210.0f;
+            }
+
+            if (this->mainAlpha >= 210) {
+                this->subAlpha += VREG(6);
+                if (this->subAlpha >= 255.0f) {
+                    this->subAlpha = 255.0f;
+                }
+
+                if (this->subAlpha >= 200) {
+                    this->copyrightAlpha += this->copyrightAlphaStep;
+                    if (this->copyrightAlpha >= 255.0f) {
+                        this->copyrightAlpha = 255.0f;
+                        this->globalState = MAG_STATE_DISPLAY;
+                        sDelayTimer = 20;
+                    }
+                }
+            }
+        }
+    } else if (this->globalState == MAG_STATE_FADE_OUT) {
+        this->effectAlpha -= this->fadeOutAlphaStep;
```

Comment:
I prefer step in general, and it's what we usually use for that. As usual, that can be discussed and we may want to change it, but in that case it would be better to standardize it globally on its own.